arwen/elrondapi/bigintOps.go

```diff
@@ -574,9 +575,10 @@ func bigIntTMod(context unsafe.Pointer, destination, op1, op2 int32) {
 	dest, a, b := bigInt.GetThree(destination, op1, op2)
 	if b.Sign() == 0 {
 		runtime := arwen.GetRuntimeContext(context)
-		runtime.SignalUserError("[bigIntTMod] division by 0")
+		runtime.SignalUserError(arwen.UserErrorDivZero)
+	} else {
+		dest.Rem(a, b) // Rem implements truncated modulus (like Go)
```

Comment:
no need for else. call return after signalusererror

Reply:
the same for all below

Reply:
same as above, arwen crashes with panic

Reply:
Move gas consumption to be the first operation of the method, before any opportunity to exit with error. That way you can `return` immediately after `runtime.SignalUserError()`.